src/Services/QueuedJobService.php

```diff
@@ -1533,6 +1579,271 @@ private function addJobHandlersToLogger(LoggerInterface $logger, QueuedJob $job,
+            return;
+        }
+
+        $jobsToRetry = $autoRetryJobs->byIDs($retryJobIDs);
+
+        /** @var QueuedJobDescriptor $jobDescriptor */
+        foreach ($jobsToRetry as $jobDescriptor) {
+            $jobClass = $jobDescriptor->Implementation;
+            $retryConfig = array_key_exists($jobClass, $jobConfigs)
+                ? $jobConfigs[$jobClass]
+                : [];
+
+            // Couldn't find a retry config for this job (possible edge case in parallel processing)
+            if (!$retryConfig) {
+                continue;
+            }
+
+            [
+                $maxRetryAttempts,
+                $initialRetryDelay,
+                $retryFalloffMultiplier,
+                $retryFalloffMultiplierVariance,
+            ] = $retryConfig;
+
+            // Determine the config key that is currently active
+            $retryCount = max(0, $jobDescriptor->RetryCount);
+
+            // We have reached the maximum retry attempts for this job
+            if ($retryCount >= $maxRetryAttempts) {
+                continue;
+            }
+
+            /** @var QueuedJob $job */
+            $job = Injector::inst()->create($jobClass);
+
+            // Invalid implementation (possible edge case of invalid data)
+            if (!$job instanceof QueuedJob) {
+                continue;
+            }
+
+            if ($initialRetryDelay > 0) {
+                // Initial delay is configured - calculate the real delay value for this retry attempt
+
+                // We will use initial delay as our starting value
+                $delay = $initialRetryDelay;
+
+                // Calculate multiplier factoring in spread
+                $multiplier = $this->getJobRetryDelayMultiplier(
+                    $retryFalloffMultiplier,
+                    $retryFalloffMultiplierVariance
+                );
+
+                // Apply multiplier to total delay, factor in number of retries
+                $delay *= pow($multiplier, $retryCount);
+                $delay = (int) floor($delay);
+            } else {
+                // Initial delay is missing - retry right away
+                $delay = 0;
+            }
+
+            $dateMessageSegment = 'immediately';
+
+            // Add a delay if configured, so we wouldn't retry the job immediately, but with some spread
+            // this lowers the chance of multiple jobs accessing the same data, thus preventing conflicts
+            if ($delay > 0) {
+                $scheduledTime = $now->getTimestamp() + $delay;
+                $jobDescriptor->StartAfter = $scheduledTime;
+
+                $formattedDate = DBDatetime::create_field('Datetime', $scheduledTime);
+                $dateMessageSegment = sprintf('after %s', $formattedDate->Rfc2822());
+            }
+
+            // Capture exception in the messages of the stuck job for better debug options
+            $message = sprintf('Automatic job retry - job will retry %s', $dateMessageSegment);
```

Comment:
Other messages stored through this service use `_t()` so we should do so here as well.